src/writer/titleListingHandler.cpp

```diff
@@ -93,24 +112,11 @@ void TitleListingHandler::handle(Dirent* dirent, std::shared_ptr<Item> item)
 void TitleListingHandler::handle(Dirent* dirent, const Hints& hints)
 {
   m_dirents.push_back(dirent);
-}
-
-DirentHandler::Dirents TitleListingHandlerV1::createDirents() const {
-  Dirents ret;
-  if (!m_dirents.empty()) {
-    ret.push_back(mp_creatorData->createDirent(NS::X, "listing/titleOrdered/v1", "application/octet-stream+zimlisting", ""));
-  }
-  return ret;
-}
-
-void TitleListingHandlerV1::handle(Dirent* dirent, const Hints& hints)
-{
-  bool isFront { false };
   try {
-    isFront = bool(hints.at(FRONT_ARTICLE));
+    if(bool(hints.at(FRONT_ARTICLE))) {
+      m_hasFrontArticles = true;
```

Comment:
There is one subtle problem with this new data member. What if all front articles are removed in `TitleListingHandler::stop()`? I understand that it is an unrealistic corner case, but it occurred to me anyway.

Reply:
This is a complex problem.
We are calling `createDirent` before we remove dirents and before we call `TitleListingHandle::stop()`. So, when we create the handler's dirents, we don't know if we will remove all frontArticle.

If this is the case, we will create a empty V1 listing. This is something we've try to avoid to simplify the (other) reader implementation but it is still correct regarding the specs. I think we can keep this as it is as it is a unrealistic corner case.